### PR TITLE
Implemented correct answers to TSourceEngineQuery server queries

### PR DESCRIPTION
**Pull request by jeefo**
Corrected answer to A2S_INFO, and implemented A2S_RULES, A2S_PLAYER,   ~~A2A_PING~~ 

It's also fixes alot of problems with third-party server query stuff like SteamCondenser, Hlstats, HLSW etc.

was:
image

became:
image

### CODE REVIEW

File: engine/server/sv_client.c [hunk 1]
```diff
@@ -3150,7 +3092,8 @@ void SV_ConnectionlessPacket( netadr_t from, sizebuf_t *msg )
 	pcmd = Cmd_Argv( 0 );
 	Con_Reportf( "SV_ConnectionlessPacket: %s : %s\n", NET_AdrToString( from ), pcmd );
 
-	if( !Q_strcmp( pcmd, "ping" )) SV_Ping( from );
+	if (SV_SourceQuery_HandleConnnectionlessPacket ( pcmd, from ) ) return;
```
Comment (a1batross) [inline engine/server/sv_client.c:3095]: IMHO it should first check our packets, and then check Source ones.

Source packets have single letter header, and it clashes with our packets. Like `i` vs `info`.

Comment (jeefo) [reply engine/server/sv_client.c:3095]: Agree about ``info`` but...

goldsrc-style A2A_PING really used anywhere now inside xash ? Probably it should be dropped in favor of actual goldsrc (source-style) query protocol ?

Comment (a1batross) [reply engine/server/sv_client.c]: These packets are still the part of the protocol.

A2A_PING would be nice to use in the next ServerBrowser iteration with accurate ping times, detailed server info and other, but then I would use Xash's own protocol anyway.

Comment (jeefo) [reply engine/server/sv_client.c]: okay, returned back old-style A2A_PING, source-query now doesn't handle these type of packet.

**Comment by a1batross** [comment]
It's fine by me. I will squash merge it, and fix the styling issues myself.

**Comment by a1batross** [comment]
Merged now.

**Pull request closed without merging**